### ISSUE TITLE
How should we allow users to edit an outstanding form submission?

### ISSUE THREAD

**Issue by flora8984461** [EclipseFdn/react-eclipsefdn-members#37]
Regarding "existing" forms. The consensus seems to be: The two options should be 1) start a new form or 2) edit the existing one (if there is one one file) 

We shouldn't need to ask that question if we know there is no existing form on file.

**Comment by flora8984461** [comment]
So if there are no "existing" forms, we directly direct the user to the first step right after login.

**Comment by flora8984461** [comment]
Question: Can they have multiple existing forms?

**Comment by chrisguindon** [comment]
> Question: Can they have multiple existing forms?

Nope! We plan to allow a user to continue editing their previous form or start a new one.

**Comment by flora8984461** [comment]
> > Question: Can they have multiple existing forms?
> 
> Nope! We plan to allow a user to continue editing their previous form or start a new one.

Sorry forgot to ask, so in this case, if a user already has one form and he chose to start a new one, it overrides the old existing form, right? 🤔

**Comment by chrisguindon** [comment]
@flora8984461 That would be the idea, yes!

**Comment by flora8984461** [comment]
Notes:
When creating a new form when the user has an existing form, it generates an empty form, once he hits "Next", I send a `POST` to `/form` and get the returned new form_Id to use for next steps. (For my end, I use a context to store the form_Id)

**Comment by autumnfound** [comment]
> Question & notes:
> When creating a new form when the user has an existing form, it generates an empty form, once he hits "Next", I send a `POST` to `/form` and get the returned new form_Id to use for next steps. (For my end, I use a context to store the form_Id)
> 
> And at the same time need to send a DELETE to delete the previous existing form. And update the association with the user.
> 
> Or
> 
> I should use the same previous form_id then replace it with the new content? Then how to clear the data in the server for the next steps?
> 
> * I can track the user is creating a new form, and store this status in Context, and don't fetch data to keep each step empty when the first load. In this way, we can use the same id.
> 
> Please feel free to discuss if my thought is good...
> 
> Question is:
> When creating a new form when having existing, do we delete the old form_id and POST a new form_id?
> Or Do we use the same form_id and just replacing the contents?

In terms of the flow of data, the best way to maintain our data integrity is to just delete the form and have it cascade, and take care of all of the downstream data. This way one call would clean up all of the form data, and then a new form can be created. Otherwise, you might need to go in and clean out all of the data for all of the other form elements (contacts, working groups, organization) on new form start. In the system, I don't think we necessarily need to restrict that data. We can just add a date_created field and you can grab the latest form if there are multiple.

This will add some fluff to our database if they don't get cleaned up immediately, but it has a good side effect. If someone wants to recover their form data, they can if it's within 30 days.

TLDR: New form, and probably call to delete old form.

**Comment by chrisguindon** [comment]
From a technical side:

The user only has 2 options:

1) Edit an existing form
2) Start a new one

I think the issue for us right now, is how should we allow our users to do that. 

@shanda-eclipse @meKokabi 

I think we need to ask this question to the end-user after they log in. However, I am open to suggestions!

**Comment by chrisguindon** [comment]
Shanda this post this comment about this here:
https://github.com/EclipseFdn/react-eclipsefdn-members/issues/17#issuecomment-790050720

**Comment by chrisguindon** [comment]
Done!
